spaceship/lib/spaceship/tunes/iap_detail.rb

```diff
@@ -0,0 +1,185 @@
+          parsed_intervals << {
+            tier: interval["value"]["tierStem"].to_i,
+            begin_date: interval["value"]["priceTierEffectiveDate"],
+            end_date: interval["value"]["priceTierEndDate"],
+            grandfathered: interval["value"]["grandfathered"],
+            country: interval["value"]["country"]
+          }
+        end
+        return parsed_intervals
+      end
+
+      # @return (String) Human Readable type of the purchase
+      def type
+        Tunes::IAPType.get_from_string(raw_data["addOnType"])
+      end
+
+      # @return (String) Human Readable status of the purchase
+      def status
+        Tunes::IAPStatus.get_from_string(raw_data["versions"].first["status"])
+      end
+
+      # Saves the current In-App-Purchase
+      def save!
+        # Transform localization versions back to original format.
+        versions_array = []
+        versions.each do |language, value|
+          versions_array << {
+                    value: {
+                      description: { value: value[:description] },
+                      name: { value: value[:name] },
+                      localeCode: language.to_s
+                    }
+          }
+        end
+
+        raw_data.set(["versions"], [{ reviewNotes: @review_notes, details: { value: versions_array } }])
+
+        # transform pricingDetails
+        intervals_array = []
+        pricing_intervals.each do |interval|
+          intervals_array << {
+            value: {
+              tierStem: interval[:tier],
+              priceTierEffectiveDate: interval[:begin_date],
+              priceTierEndDate: interval[:end_date],
+              country: interval[:country] || "WW",
+              grandfathered: interval[:grandfathered]
+            }
+          }
+        end
+
+        if subscription_price_target
+          intervals_array = []
+          pricing_calculator = client.iap_subscription_pricing_target(app_id: application.apple_id, purchase_id: purchase_id, currency: subscription_price_target[:currency], tier: subscription_price_target[:tier])
+          pricing_calculator.each do |language_code, value|
+            intervals_array << {
+              value: {
+                tierStem: value["tierStem"],
+                priceTierEffectiveDate: value["priceTierEffectiveDate"],
+                priceTierEndDate: value["priceTierEndDate"],
+                country: language_code,
+                grandfathered: { value: "FUTURE_NONE" }
+              }
+            }
+          end
+
+        end
+
+        raw_data.set(["pricingIntervals"], intervals_array)
+
+        if @review_screenshot
+          # Upload Screenshot
+          upload_file = UploadFile.from_path @review_screenshot
+          screenshot_data = client.upload_purchase_review_screenshot(application.apple_id, upload_file)
+          new_screenshot = {
+            "value" => {
+              "assetToken" => screenshot_data["token"],
```

Comment:
sadly we have no AppVersion in the IAP context, are cool if do a refactor on this after iap is merged?
moving the `generate_image_metadata`  to fastlane-core - and then find all cases where it could be used